tests/test_noise.py

```diff
@@ -130,3 +138,201 @@ def pinkish_spectrum(x, y):
     # Check we are achieving an estimate near the target
     logger.debug(f"Estimated Noise Variance {estimated_noise_var}")
     assert np.isclose(estimated_noise_var, target_noise_variance, rtol=0.1)
+
+    # Check sampling yields an estimate near target.
+    sample_n = 16
+    sample_res = 32
+    im_zeros = Image(np.zeros((sample_n, sample_res, sample_res)))
+    im_noise_sample = sim_fixture.noise_adder._forward(im_zeros, range(sample_n))
+    sampled_noise_var = np.var(im_noise_sample.asnumpy())
+
+    logger.debug(f"Sampled Noise Variance {sampled_noise_var}")
+    assert np.isclose(sampled_noise_var, target_noise_variance, rtol=0.1)
+
+
+@pytest.mark.parametrize("res", RESOLUTIONS, ids=lambda param: f"res={param}")
+@pytest.mark.parametrize(
+    "target_noise_variance", VARS, ids=lambda param: f"var={param}"
+)
+@pytest.mark.parametrize("dtype", DTYPES, ids=lambda param: f"dtype={param}")
```

Comment:
This is nice. I like this as opposed to having a product of parameters.

Reply:
Cool.  I think they both have times they work out, but generally i'm happier with the code we moved towards pytest.